### PR TITLE
SNC: 30 April

### PR DESCRIPTION
**Pull request by Northmoc**
Some more cards that didn't need engine support
Structural Assault stack description for the DamageAll line is kind of bad, I wanted to look at that for Aggravate anyway - that would be a different PR though
Some scripts from @TheBlackMarvel too but I will admit to editing them pretty heavily 😅

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/structural_assault.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:Structural Assault
+ManaCost:3 R R
+Types:Sorcery
+A:SP$ DestroyAll | ValidCards$ Artifact | SubAbility$ DBDamageAll | SpellDescription$ Destroy all artifacts,
+SVar:DBDamageAll:DB$ DamageAll | NumDmg$ X | ValidCards$ Creature | ValidDescription$ each creature | SpellDescription$ then CARDNAME deals damage to each creature equal to the number of artifacts that were put into graveyards from the battlefield this turn.
+SVar:X:Count$ThisTurnEntered_Graveyard_from_Battlefield_Card.Artifact
+AI:RemoveDeck:Random
+Destroy all artifacts, then Structural Assault deals damage to each creature equal to the number of artifacts that were put into graveyards from the battlefield this turn.
```
Comment (Agetian) [inline forge-gui/res/cardsfolder/upcoming/structural_assault.txt:8]: The Oracle: tag is missing

File: forge-gui/res/cardsfolder/upcoming/soul_of_emancipation.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Name:Soul of Emancipation
+ManaCost:4 G W U
+Types:Creature Avatar
+PT:5/7
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDestroy | TriggerDescription$ When CARDNAME enters the battlefield, destroy up to three other target nonland permanents. For each of those permanents, its controller creates a 3/3 white Angel creature token with flying.
+SVar:TrigDestroy:DB$ Destroy | TargetMin$ 0 | TargetMax$ 3 | ValidTgts$ Permanent.nonLand | TgtPrompt$ Choose up to three nonland permanents | RememberTargets$ True | SubAbility$ MakeTokens
+SVar:MakeTokens:DB$ RepeatEach | RepeatSubAbility$ DBToken | DefinedCards$ Remembered | UseImprinted$ True | ChangeZoneTable$ True
+SVar:DBToken:DB$ Token | TokenScript$ w_3_3_angel_flying | TokenOwner$ ImprintedController
+DeckHints:Ability$Token & Type$Angel
```
Comment (Agetian) [inline forge-gui/res/cardsfolder/upcoming/soul_of_emancipation.txt:9]: Not sure if the Token ability hint is truly necessary here... What's the rationale for it? This card doesn't seem to have an obvious synergy with other token-generating cards...? I'm probably missing something though :D

File: forge-gui/res/cardsfolder/upcoming/backstreet_bruiser.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Name:Backstreet Bruiser
+ManaCost:1 U
+Types:Creature Cephalid Rogue
+PT:3/3
+K:Defender
+S:Mode$ Continuous | Affected$ Card.Self | AddHiddenKeyword$ CARDNAME can attack as though it didn't have defender. | CheckSVar$ X | SVarCompare$ GE2 | Description$ As long as there are two or more counters among creatures you control, CARDNAME can attack as though it didn't have defender.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/backstreet_bruiser.txt:6]: IsPresent maybe?

File: forge-gui/res/cardsfolder/upcoming/soul_of_emancipation.txt (hunk 2 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/soul_of_emancipation.txt:9]: Nah, just came wrong and I missed that it was Hints instead of Has 🤦

**Review by Agetian — APPROVED** [review]

**Review by paulsnoops — APPROVED** [review]

**Pull request merged**